From 6567024fb6241bfbf851a3f5a933fc1ba633bac7 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Fri, 23 Aug 2019 23:49:56 +0200
Subject: [PATCH] fix pkg_check JsonCpp and hdhomerun

---
 FindJsonCpp.cmake   | 14 +++++++-------
 Findhdhomerun.cmake | 30 +++++++++++++++---------------
 2 files changed, 22 insertions(+), 22 deletions(-)

diff --git a/FindJsonCpp.cmake b/FindJsonCpp.cmake
index bdada8d..8281d62 100644
--- a/FindJsonCpp.cmake
+++ b/FindJsonCpp.cmake
@@ -1,15 +1,15 @@
 find_package(PkgConfig)
 if(PKG_CONFIG_FOUND)
-  pkg_check_modules (JSONCPP jsoncpp)
+  pkg_check_modules(PC_JSONCPP jsoncpp QUIET)
 endif()
 
-if(NOT JSONCPP_FOUND)
-  find_path(JSONCPP_INCLUDE_DIRS json/json.h
-            PATH_SUFFIXES jsoncpp)
-  find_library(JSONCPP_LIBRARIES jsoncpp)
-endif()
+find_path(JSONCPP_INCLUDE_DIRS json/json.h
+                               PATHS ${PC_JSONCPP_INCLUDEDIR}
+                               PATH_SUFFIXES jsoncpp)
+find_library(JSONCPP_LIBRARIES jsoncpp
+                               PATHS ${PC_JSONCPP_LIBDIR})
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(JsonCpp DEFAULT_MSG JSONCPP_LIBRARIES JSONCPP_INCLUDE_DIRS)
+find_package_handle_standard_args(JsonCpp REQUIRED_VARS JSONCPP_LIBRARIES JSONCPP_INCLUDE_DIRS)
 
 mark_as_advanced(JSONCPP_INCLUDE_DIRS JSONCPP_LIBRARIES)
diff --git a/Findhdhomerun.cmake b/Findhdhomerun.cmake
index 716b35c..1b9d884 100644
--- a/Findhdhomerun.cmake
+++ b/Findhdhomerun.cmake
@@ -1,15 +1,15 @@
-find_package(PkgConfig)
-if(PKG_CONFIG_FOUND)
-  pkg_check_modules (HDHOMERUN hdhomerun)
-endif()
-
-if(NOT HDHOMERUN_FOUND)
-  find_path(HDHOMERUN_INCLUDE_DIRS hdhomerun.h
-            PATH_SUFFIXES hdhomerun libhdhomerun)
-  find_library(HDHOMERUN_LIBRARIES hdhomerun)
-endif()
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(hdhomerun DEFAULT_MSG HDHOMERUN_LIBRARIES HDHOMERUN_INCLUDE_DIRS)
-
-mark_as_advanced(HDHOMERUN_INCLUDE_DIRS HDHOMERUN_LIBRARIES)
+find_package(PkgConfig)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(PC_HDHOMERUN hdhomerun QUIET)
+endif()
+
+find_path(HDHOMERUN_INCLUDE_DIRS hdhomerun.h
+                                 PATHS ${PC_HDHOMERUN_INCLUDEDIR}
+                                 PATH_SUFFIXES hdhomerun libhdhomerun)
+find_library(HDHOMERUN_LIBRARIES hdhomerun
+                                 PATHS ${PC_HDHOMERUN_LIBDIR})
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(hdhomerun REQUIRED_VARS HDHOMERUN_LIBRARIES HDHOMERUN_INCLUDE_DIRS)
+
+mark_as_advanced(HDHOMERUN_INCLUDE_DIRS HDHOMERUN_LIBRARIES)
